### ISSUE TITLE
[Bug] Whitespace Padding in Process Command Line rule matches every process

### ISSUE THREAD

**Issue by peterhirn** [elastic/detection-rules#1641]
**Describe the bug**

Whitespace Padding in Process Command Line rule matches every process, regardless of whitespaces.

**To Reproduce**
- Elastic Cloud
- Import and enable predefined rules
- Install Elastic Agent on Windows 10

Sample match

```
CommandLine: C:\WINDOWS\system32\cmd.exe /c CALL "C:\Program Files\nodejs\\node.exe" "C:\Program Files\nodejs\\node_modules\npm\bin\npm-cli.js" prefix -g
```

**Comment by w0rk3r** [comment]
Hey @peterhirn, I tried to reproduce this, and I wasn't able to do so. Can you please provide a sample alert on this?

You can get the JSON by:

Security > Alerts > Go to the alert > View Details > JSON

Make sure to remove/modify any confidential/sensitive information before sharing it here.

**Comment by peterhirn** [comment]
I just realized that all alerts are coming from `windows.sysmon_operational`. I didn't mention that I have sysmon installed using this configuration: https://github.com/SwiftOnSecurity/sysmon-config

Please give me a heads up if you discover PII or confidential info in the json.

The rule now also contains a custom webhook action. I've been using it to test alerting, as it is triggered constantly.

```
{
  "_id": "6a380cc52a38da3124bffd75ea7476eb57792b258efce2f180d84e21ae510903",
  "_index": ".siem-signals-default-000001",
  "_score": "1",
  "_type": "_doc",
  "@timestamp": "2021-12-02T15:21:56.841Z",
  "agent": {
    "ephemeral_id": "22a44169-9e1b-471c-9833-57e645bf7a5a",
    "hostname": "peter",
    "id": "125cbe0c-bffc-48c5-a596-c4ce1c2beec1",
    "name": "peter",
    "type": "filebeat",
    "version": "7.15.2"
  },
  "data_stream": {
    "dataset": "windows.sysmon_operational",
    "namespace": "default",
    "type": "logs"
  },
  "ecs": {
    "version": "1.12.0"
  },
  "elastic_agent": {
    "id": "125cbe0c-bffc-48c5-a596-c4ce1c2beec1",
    "snapshot": "false",
    "version": "7.15.2"
  },
  "event": {
    "action": "Process Create (rule: ProcessCreate)",
    "agent_id_status": "verified",
    "category": "process",
    "code": "1",
    "created": "2021-12-02T15:21:46.181Z",
    "dataset": "windows.sysmon_operational",
    "ingested": "2021-12-02T15:21:47.000Z",
    "kind": "signal",
    "module": "windows",
    "provider": "Microsoft-Windows-Sysmon",
    "type": "start"
  },
  "host": {
    "architecture": "x86_64",
    "hostname": "peter",
    "id": "<redacted>",
    "ip": "<redacted>",
    "mac": "<redacted>",
    "name": "peter",
    "os": {
      "build": "19041.1348",
      "family": "windows",
      "kernel": "10.0.19041.1348 (WinBuild.160101.0800)",
      "name": "Windows 10 Enterprise",
      "platform": "windows",
      "type": "windows",
      "version": "10.0"
    }
  },
  "input": {
    "type": "winlog"
  },
  "kibana": {
    "alert": {
      "ancestors": {
        "depth": "0",
        "id": "RIK8e30BYedMAoKLv_TG",
        "index": ".ds-logs-windows.sysmon_operational-default-2021.11.30-000001",
        "type": "event"
      },
      "depth": "1",
      "original_event": {
        "action": "Process Create (rule: ProcessCreate)",
        "category": "process",
        "code": "1",
        "created": "2021-12-02T15:21:46.181Z",
        "dataset": "windows.sysmon_operational",
        "kind": "event",
        "module": "windows",
        "provider": "Microsoft-Windows-Sysmon",
        "type": "start"
      },
      "original_time": "2021-12-02T15:21:44.157Z",
      "reason": "process event with process svchost.exe, parent process services.exe, by SYSTEM on peter created medium alert Whitespace Padding in Process Command Line.",
      "risk_score": "47",
      "rule": {
        "author": "Elastic",
        "created_at": "2021-11-29T16:41:58.003Z",
        "created_by": "3007801101",
        "description": "Identifies process execution events where the command line value contains a long sequence of whitespace characters or multiple occurrences of contiguous whitespace. Attackers may attempt to evade signature-based detections by padding their malicious command with unnecessary whitespace characters. These observations should be investigated for malicious behavior.",
        "enabled": "true",
        "from": "now-540s",
        "immutable": "true",
        "index": "winlogbeat-*,logs-endpoint.events.*,logs-windows.*",
        "interval": "5m",
        "language": "eql",
        "license": "Elastic License v2",
        "max_signals": "100",
        "name": "Whitespace Padding in Process Command Line",
        "note": "## Triage and analysis\n\n- Analyze the command line of the process in question for evidence of malicious code execution.\n- Review the ancestor and child processes spawned by the process in question for indicators of further malicious code execution.",
        "query": "process where event.type in (\"start\", \"process_started\") and\n  process.command_line regex \".*[ ]{20,}.*\" or \n  \n  /* this will match on 3 or more separate occurrences of 5+ contiguous whitespace characters */\n  process.command_line regex \".*(.*[ ]{5,}[^ ]*){3,}.*\"\n",
        "references": "https://twitter.com/JohnLaTwC/status/1419251082736201737",
        "rule_id": "e0dacebe-4311-4d50-9387-b17e89c2e7fd",
        "tags": "Elastic,Host,Windows,Threat Detection,Defense Evasion",
        "threat": {
          "framework": "MITRE ATT&CK",
          "tactic": {
            "id": "TA0005",
            "name": "Defense Evasion",
            "reference": "https://attack.mitre.org/tactics/TA0005/"
          }
        },
        "to": "now",
        "type": "eql",
        "updated_at": "2021-12-02T15:16:57.062Z",
        "updated_by": "3007801101",
        "uuid": "43fda112-5133-11ec-9a84-457989ab86ac",
        "version": "2"
      },
      "severity": "medium",
      "workflow_status": "open"
    }
  },
  "log": {
    "level": "information"
  },
  "message": "<redacted>",
  "process": {
    "args": "C:\\WINDOWS\\system32\\svchost.exe,-k,netsvcs,-p,-s,gpsvc",
    "args_count": "6",
    "command_line": "C:\\WINDOWS\\system32\\svchost.exe -k netsvcs -p -s gpsvc",
    "entity_id": "{4938033b-e488-61a8-3945-050000005100}",
    "executable": "C:\\Windows\\System32\\svchost.exe",
    "hash": {
      "md5": "f586835082f632dc8d9404d83bc16316",
      "sha256": "643ec58e82e0272c97c2a59f6020970d881af19c0ad5029db9c958c13b6558c7"
    },
    "name": "svchost.exe",
    "parent": {
      "args": "C:\\WINDOWS\\system32\\services.exe",
      "args_count": "1",
      "command_line": "C:\\WINDOWS\\system32\\services.exe",
      "entity_id": "{4938033b-842b-6197-0b00-000000005100}",
      "executable": "C:\\Windows\\System32\\services.exe",
      "name": "services.exe",
      "pid": "1056"
    },
    "pe": {
      "company": "Microsoft Corporation",
      "description": "Host Process for Windows Services",
      "file_version": "10.0.19041.546 (WinBuild.160101.0800)",
      "imphash": "f9bbd96fae53b7a31264a703cafa0666",
      "original_file_name": "svchost.exe",
      "product": "Microsoft® Windows® Operating System"
    },
    "pid": "5668",
    "working_directory": "C:\\WINDOWS\\system32\\"
  },
  "related": {
    "hash": "643ec58e82e0272c97c2a59f6020970d881af19c0ad5029db9c958c13b6558c7,f586835082f632dc8d9404d83bc16316,f9bbd96fae53b7a31264a703cafa0666",
    "user": "SYSTEM"
  },
  "signal": {
    "_meta": {
      "version": "57"
    },
    "ancestors": "{\"id\":\"RIK8e30BYedMAoKLv_TG\",\"type\":\"event\",\"index\":\".ds-logs-windows.sysmon_operational-default-2021.11.30-000001\",\"depth\":0}",
    "depth": "1",
    "original_event": {
      "action": "Process Create (rule: ProcessCreate)",
      "agent_id_status": "verified",
      "category": "process",
      "code": "1",
      "created": "2021-12-02T15:21:46.181Z",
      "dataset": "windows.sysmon_operational",
      "ingested": "2021-12-02T15:21:47Z",
      "kind": "event",
      "module": "windows",
      "provider": "Microsoft-Windows-Sysmon",
      "type": "start"
    },
    "original_time": "2021-12-02T15:21:44.157Z",
    "parent": {
      "depth": "0",
      "id": "RIK8e30BYedMAoKLv_TG",
      "index": ".ds-logs-windows.sysmon_operational-default-2021.11.30-000001",
      "type": "event"
    },
    "parents": "{\"id\":\"RIK8e30BYedMAoKLv_TG\",\"type\":\"event\",\"index\":\".ds-logs-windows.sysmon_operational-default-2021.11.30-000001\",\"depth\":0}",
    "reason": "process event with process svchost.exe, parent process services.exe, by SYSTEM on peter created medium alert Whitespace Padding in Process Command Line.",
    "rule": {
      "actions": "{\"group\":\"default\",\"params\":{\"body\":\"[{{context.alerts}}]\"},\"actionTypeId\":\".webhook\",\"actionRef\":\"action_0\"}",
      "author": "Elastic",
      "created_at": "2021-11-29T16:41:58.003Z",
      "created_by": "3007801101",
      "description": "Identifies process execution events where the command line value contains a long sequence of whitespace characters or multiple occurrences of contiguous whitespace. Attackers may attempt to evade signature-based detections by padding their malicious command with unnecessary whitespace characters. These observations should be investigated for malicious behavior.",
      "enabled": "true",
      "exceptions_list": "",
      "false_positives": "",
      "filters": "",
      "from": "now-540s",
      "id": "43fda112-5133-11ec-9a84-457989ab86ac",
      "immutable": "true",
      "index": "winlogbeat-*,logs-endpoint.events.*,logs-windows.*",
      "interval": "5m",
      "language": "eql",
      "license": "Elastic License v2",
      "max_signals": "100",
      "meta": {
        "from": "4m",
        "kibana_siem_app_url": "<redacted>"
      },
      "name": "Whitespace Padding in Process Command Line",
      "note": "## Triage and analysis\n\n- Analyze the command line of the process in question for evidence of malicious code execution.\n- Review the ancestor and child processes spawned by the process in question for indicators of further malicious code execution.",
      "output_index": ".siem-signals-default",
      "query": "process where event.type in (\"start\", \"process_started\") and\n  process.command_line regex \".*[ ]{20,}.*\" or \n  \n  /* this will match on 3 or more separate occurrences of 5+ contiguous whitespace characters */\n  process.command_line regex \".*(.*[ ]{5,}[^ ]*){3,}.*\"\n",
      "references": "https://twitter.com/JohnLaTwC/status/1419251082736201737",
      "risk_score": "47",
      "risk_score_mapping": "",
      "rule_id": "e0dacebe-4311-4d50-9387-b17e89c2e7fd",
      "severity": "medium",
      "severity_mapping": "",
      "tags": "Elastic,Host,Windows,Threat Detection,Defense Evasion",
      "threat": "{\"framework\":\"MITRE ATT&CK\",\"tactic\":{\"reference\":\"https://attack.mitre.org/tactics/TA0005/\",\"name\":\"Defense Evasion\",\"id\":\"TA0005\"},\"technique\":[]}",
      "timestamp_override": "event.ingested",
      "to": "now",
      "type": "eql",
      "updated_at": "2021-12-02T15:16:57.062Z",
      "updated_by": "3007801101",
      "version": "2"
    },
    "status": "open"
  },
  "user": {
    "domain": "NT AUTHORITY",
    "id": "S-1-5-18",
    "name": "SYSTEM"
  },
  "winlog": {
    "api": "wineventlog",
    "channel": "Microsoft-Windows-Sysmon/Operational",
    "computer_name": "peter",
    "event_data": {
      "Company": "Microsoft Corporation",
      "Description": "Host Process for Windows Services",
      "FileVersion": "10.0.19041.546 (WinBuild.160101.0800)",
      "IntegrityLevel": "System",
      "LogonGuid": "<redacted>",
      "LogonId": "0x3e7",
      "ParentUser": "NT AUTHORITY\\SYSTEM",
      "Product": "Microsoft® Windows® Operating System",
      "TerminalSessionId": "0"
    },
    "event_id": "1",
    "opcode": "Info",
    "process": {
      "pid": "21628",
      "thread": {
        "id": "22444"
      }
    },
    "provider_guid": "{5770385f-c22a-43e0-bf4c-06f5698ffbd9}",
    "provider_name": "Microsoft-Windows-Sysmon",
    "record_id": "66314",
    "task": "Process Create (rule: ProcessCreate)",
    "user": {
      "domain": "NT AUTHORITY",
      "identifier": "S-1-5-18",
      "name": "SYSTEM",
      "type": "User"
    },
    "version": "5"
  }
}
```

**Comment by w0rk3r** [comment]
@peterhirn I've installed sysmon in a new VM and tried to reproduce the above, but still no success...

As an test, can you do the following:

1- Open the Rule
2- Copy the query
3- Go to Security > Timelines > Create New Timeline > Correlation
4- Paste the EQL query and expand the time

The query still match on the events?

**Comment by peterhirn** [comment]
This is really confusing, but great that I learn how to use Kibana with your instructions ;)

Yes, it still matches, even after removing the second half of the EQL:

```
process where event.type in ("start", "process_started") and process.command_line regex ".*[ ]{20,}.*"
```

I copy-pasted this exact regex to https://regex101.com/ and it works as expected.

It seems irrelevant how many spaces I match, eg. `".*[ ]{1000,}.*"` same result. I tested some other regexes and they seem to work fine `".*git\\.exe.*"`.

**Comment by w0rk3r** [comment]
Hey @peterhirn, thanks for bringing this one to our attention!

Actually, after talking with my teammates, seems that we already saw this kind of bug with the same rule, which was discussed and solved in this [issue](https://github.com/elastic/elasticsearch/issues/78391).

The problem:

Sysmon uses the wildcard field type for the command_line field, which in the versions < 7.16 have this bug that make regex match unexpected results.

The solution:

As this is a fixed bug in 7.16, we are removing the support for data sources other than endpoint events (which uses text field type instead of wildcard) in versions prior to 7.16, and continue to support them in versions >= 7.16 using the `min_stack_version` field, which makes new changes to this rule to not be backported to stack versions that are affected by this bug.

First PR - #1647 
Final PR - #1648
